test/unit/gotcha_unit_tests.c

```diff
@@ -274,56 +274,27 @@ Suite* gotcha_auxv_suite(){
 ////////////////////////////////////////////////////GOTCHA Hash Tests///////////////////////////////////////////////////////////////////////////////////
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-START_TEST(hash_insert_test){
-   hash_table_t table;
-   create_hashtable(&table,50,(hash_func_t) strhash, (hash_cmp_t) gotcha_strcmp);
-   const char* hashable_string = "dogs";
-   int ins_return=-1,look_return =-1, del_return=-1;
-   hash_hashvalue_t dogs_val = strhash(hashable_string);  
-   struct hash_entry_t* lookup_holder;;
-   ins_return=insert(&table,hashable_string,(void*)1337,dogs_val);
-   ck_assert_msg(!ins_return,"Internal error in insert function");
-   look_return = lookup(&table,hashable_string,&lookup_holder);
-   int found_value = extract_data(lookup_holder);
-   ck_assert_msg(!look_return,"Internal error in lookup function");
-   del_return = removefrom_hashtable(&table,hashable_string);
-   ck_assert_msg(!del_return,"Internal error in element removal function");
-   ck_assert_msg((int)found_value==1337,"Inserted item could not be looked up in hash table");
-   ck_assert_msg(!destroy_hashtable(&table),"Could not destroy hashtable");
-}
-END_TEST
+#define NUM_INSERTS 8192
+#define TABLE_SIZE (NUM_INSERTS/2)
```

Comment:
This is a test, might as well abuse the code and make TABLE_SIZE start very small so it repeatedly grows